ndonnx/additional/_additional.py

```diff
@@ -139,8 +139,9 @@ def make_nullable(x: Array, null: Array) -> Array:
     TypeError
         If the data type of ``x`` does not have a nullable counterpart.
     """
-    x = ndx.asarray(x)
     out = x.dtype._ops.make_nullable(x, null)
     if out is NotImplemented:
-        raise TypeError(f"'make_nullable' not implemented for `{x.dtype}`")
+        raise ndx.UnsupportedOperationError(
```

Comment:
What is the rule here? When do we return `NotImplemented` and when do we raise `UnsupportedOperationError`? Could you summarize again the decision why we don't always rely on raising from further down the stack?

Reply:
See https://github.com/Quantco/ndonnx/pull/40#issuecomment-2275572788